### ISSUE TITLE
Add missing relocation types & architectures for ELF  

### ISSUE THREAD

**Issue by Roeegg2** [rizinorg/rizin#4699]
**Is your feature request related to a problem? Please describe.**
The code under file `librz/bin/p/bin_elf.inc` function `reloc_convert` which takes care of moving data from `RzBinElfReloc` to the general `RzBinReloc` doesn't take care of some relocation architectures (CRIS, Nios II, HPPA, and many more) and as well misses many relocation types in all of the currently implemented relocations

**Describe the solution you'd like**
Simply adding `switch`es/`case`s to `reloc_convert` take of these architectures/types

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by Roeegg2** [comment]
Same thing for `patch_reloc`. A lot types and archs aren't handled there

**Comment by XVilka** [comment]
HPPA will be covered by https://github.com/rizinorg/rizin/pull/4469

**Comment by Roeegg2** [comment]
We need to add the missing architectures and types in COFF as well... Maybe we should just unite them into one issue?
Both shouldn't take very long and aren't very hard to fix.

**Comment by themadbit** [comment]
Assign this issue to me.

**Comment by AppleMangoOrange** [comment]
I am working on this issue, but `librz/bin/p/bin_elf.inc` does not have a `reloc_convert` function. I think the code has been shifted to `Elf_(rz_bin_elf_convert_relocation)` in `librz/bin/format/elf/elf_relocs_conversion.c`. I should add a function (with a `switch-case` statement) and replace the `ARCH_MISSING` under the cases, right?

I think I need to add a case for each value under `Nios II relocations` in `elf/glib.c`, is this correct?

**Comment by notxvilka** [comment]
> I am working on this issue, but `librz/bin/p/bin_elf.inc` does not have a `reloc_convert` function. I think the code has been shifted to `Elf_(rz_bin_elf_convert_relocation)` in `librz/bin/format/elf/elf_relocs_conversion.c`. I should add a function (with a `switch-case` statement) and replace the `ARCH_MISSING` under the cases, right?
> 
> I think I need to add a case for each value under `Nios II relocations` in `elf/glib.c`, is this correct?

Precisely. We removed the nios2 architecture though because exactly zero users and no real hardware: https://github.com/rizinorg/rizin/pull/4457

**Comment by AppleMangoOrange** [comment]
The `patch_val_over_mask_32` does not check if the original add bits in the opcode are 0, and just patches the new address on the instruction: `ut32 opcode = rz_read_ble32(buf, big_endian) | rz_bits_spread(mask, val);`. Is this intended? Is it fine if I follow the same for my implementation or do I have to clear the bits before patching to make sure that the resulting opcode is correct?
I cannot use the `patch_val_over_mask_32` function in the case I am working on because AVR needs a 16-bit version of this function, which does not exist afaik.

**Comment by AppleMangoOrange** [comment]
Can I make a new `patch_val_over_mask_16` function? The AVR function is getting very long with a lot of repeated code. Should it address the above masking problem?

Or maybe there could be a general function that handles all byte values instead of hard-coding 32 and 64 bits?
